### ISSUE TITLE
gnt-instance failover and migrate does not work with manual target node selection

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#452]
Originally reported of Google Code with ID 396.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".

gnt-cluster (ganeti v2.7.0beta1-267-g4f73cfc) 2.7.0~beta1

Software version: 2.7.0~beta1
Internode protocol: 2070000
Configuration format: 2070000
OS api version: 20
Export interface: 0

<b>What distribution are you using?</b>

Debian 6.0.7

<b>What steps will reproduce the problem?</b>
1. run 'gnt-instance migrate --target-node=hostname instancename'
<b>2.</b>
<b>3.</b>

<b>What is the expected output? What do you see instead?</b>

I would expect the instance to be migrated to the specified target node. Instead I get this:

root:~# gnt-instance migrate --target-node devel0.example.com --debug inst1.example.com                            
2013-03-11 09:50:53,089: gnt-instance migrate pid=29323 cli:2411 INFO Command line: gnt-instance migrate --target-node devel0.example.com --debug inst1.example.com
Instance inst1.check24tech.de will be migrated. Note that migration
might impact the instance if anything goes wrong (e.g. due to bugs in
the hypervisor). Continue?
y/[n]/?: y
2013-03-11 09:50:55,177: gnt-instance migrate pid=29323 cli:2418 ERROR Error during command processing
Traceback (most recent call last):
  File "/usr/local/lib/python2.6/dist-packages/ganeti/cli.py", line 2414, in GenericMain
    result = func(options, args)
  File "/usr/local/lib/python2.6/dist-packages/ganeti/client/gnt_instance.py", line 799, in MigrateInstance
    SubmitOrSend(op, cl=cl, opts=opts)
  File "/usr/local/lib/python2.6/dist-packages/ganeti/cli.py", line 2185, in SubmitOrSend
    return SubmitOpCode(op, cl=cl, feedback_fn=feedback_fn, opts=opts)
  File "/usr/local/lib/python2.6/dist-packages/ganeti/cli.py", line 2162, in SubmitOpCode
    reporter=reporter)
  File "/usr/local/lib/python2.6/dist-packages/ganeti/cli.py", line 2143, in PollJob
    return GenericPollJob(job_id, _LuxiJobPollCb(cl), reporter)
  File "/usr/local/lib/python2.6/dist-packages/ganeti/cli.py", line 1965, in GenericPollJob
    errors.MaybeRaise(msg)
  File "/usr/local/lib/python2.6/dist-packages/ganeti/errors.py", line 510, in MaybeRaise
    raise errcls(*args)
OpExecError: Internal assertion error: please report this as a bug.
Error message: ''; location:
  File "/usr/local/lib/python2.6/dist-packages/ganeti/cmdlib.py", line 8651, in CheckPrereq
    assert locking.NAL in self.lu.owned_locks(locking.LEVEL_NODE_ALLOC)

Failure: command execution error:
Internal assertion error: please report this as a bug.
Error message: ''; location:
  File "/usr/local/lib/python2.6/dist-packages/ganeti/cmdlib.py", line 8651, in CheckPrereq
    assert locking.NAL in self.lu.owned_locks(locking.LEVEL_NODE_ALLOC)

<b>Please provide any additional information below.</b>

A failover instead of migrate results in the same error.

Both failover and migrate work as expected when using the hail iallocator to determine the target node.
```
Originally added on 2013-03-11 09:03:36 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Thanks for reporting this, we'll look into it and fix it for 2.7.0, plus add it to our QA.

Thanks,

Guido
```
Originally added on 2013-03-11 09:10:17 +0000 UTC.

Added Labels: Priority-High
Changed State: Accepted
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 7b3d70d47567b13a6ca5f2b5bf474c1f92a9e816
Author: Guido Trotter <ultrotter@google.com>
Date:   Tue Mar 12 16:24:44 2013 +0100

    Fix migrate/failover -n for ext mirror storage
    
    This fixes issue 396.
    
    - Fix a wrong comment that mentions drbd8 when actually the code acts
      only on externally mirrored instances.
    - Fix a wrong assert that requires failover/migrate to acquire the NAL
      on externally mirrored instances: this is the case only when a
      iallocator is used, not if a node is manually specified.
    
    Extra qa should be added, but this can only be done if an externally
    mirrored instance or a diskless instance is tested. It is easier to add
    it on the master branch, where we have diskless instances support.
    
    Signed-off-by: Guido Trotter <ultrotter@google.com>
    Reviewed-by: Iustin Pop <iustin@google.com>

```
Originally added on 2013-03-12 16:31:08 +0000 UTC.

Mentioned Issue with GoogleCode ID 396 has been migrated to GitHub with ID #452.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
